.github/workflows/main.yml

```diff
@@ -242,6 +242,36 @@ jobs:
           name: code-coverage-report-unit-aries-vcx
           path: /tmp/artifacts/coverage
 
+  #TODO - can this be included within code-coverage-aries-vcx-integration-tests?
+  code-coverage-aries-vcx-modular-dependencies-integration-tests:
+    needs: workflow-setup
+    runs-on: ubuntu-20.04
+    steps:
+      - name: "Git checkout"
+        uses: actions/checkout@v3
+      - name: "Setup rust codecov environment"
+        uses: ./.github/actions/setup-codecov-rust
+      - name: "Run workspace tests: modular_deps pool_tests agency_pool_tests"
+        run: |
+          RUSTFLAGS='-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Cpanic=abort -Zpanic_abort_tests' \
+          RUSTDOCFLAGS='-Zprofile -Ccodegen-units=1 -Cinline-threshold=0 -Clink-dead-code -Coverflow-checks=off -Cpanic=abort -Zpanic_abort_tests' \
+          RUST_TEST_THREADS=1 CARGO_INCREMENTAL=0 TEST_POOL_IP=127.0.0.1 cargo test --package aries-vcx -F 'pool_tests agency_pool_tests modular_deps';
+
+          mkdir -p /tmp/artifacts/coverage
+          grcov ./target/debug/ -s . -t lcov --llvm --branch --ignore-not-existing -o /tmp/artifacts/coverage/coverage.lcov
+      - name: "Upload coverage to Codecov"
+        uses: codecov/codecov-action@v2
+        with:
+          directory: /tmp/artifacts/coverage
+          flags: unittests-aries-vcx
+          name: codecov-unit-aries-vcx
+          fail_ci_if_error: true
+          path_to_write_report: /tmp/artifacts/coverage/codecov_report.gz
+      - uses: actions/upload-artifact@v2
+        with:
+          name: code-coverage-report-unit-aries-vcx
+          path: /tmp/artifacts/coverage
+
 #  #  ##########################################################################################
 #  #  ###############################    TESTING    ###########################################
 
```

Comment:
issue; test-android-build is not failing despite cargo test failing

Reply:
I suppose this is solved now?